src/main/java/com/opentable/server/EmbeddedJettyBase.java

```diff
@@ -305,6 +312,17 @@ public void gracefulShutdown(ContextClosedEvent evt) {
         LOG.debug("Received application context closed event {}. Shutting down...", evt);
         LOG.info("Early shutdown of Jetty connectors on {}", container);
         if (container != null) {
+            if(shouldSleepBeforeShutdown) {
+                long sleepDurationMillisBeforeShutdown = sleepDurationBeforeShutdown.toMillis();
+                LOG.info("Application config requesting sleep for {} ms before Jetty shutdown", sleepDurationMillisBeforeShutdown);
```

Comment:
Does unannounce happen before this? It needs to happen first.

Reply:
yes un-announce happens before this. see - https://github.com/opentable/service-discovery/blob/master/client/src/main/java/com/opentable/service/discovery/client/HttpServerAnnouncer.java#L186